tests/Constraint/ConstraintTest.php

```diff
@@ -158,4 +158,41 @@ public function invalidOperators()
             array('1.2.3', 'equals', 'InvalidArgumentException'),
         );
     }
-}
+
+    /**
+     * @dataProvider bounds
+     *
+     * @param string $normalizedVersion
+     * @param string|null $expectedLower
+     * @param string|null $expectedUpper
+     */
+    public function testBounds($normalizedVersion, $expectedLower, $expectedUpper)
+    {
+        $constraint = new Constraint('==', $normalizedVersion);
+
+        $this->assertSame($expectedLower, $constraint->getLowerBound());
+        $this->assertSame($expectedUpper, $constraint->getUpperBound());
+    }
+
+    /**
+     * @return array
+     */
+    public function bounds()
+    {
+        return array(
+            'none' => array('1.0.0.0', '0000001.0000000.0000000.0000000', '1999999.0999999.0999999.0999999'),
```

Comment:
Lower should be 1.0.0.0, and 1.2.3.4 on the line below 
Upper should be 1.0.0.0 here, 1.2.3.4 below